backend/api/permissions/FuelCode.py

```diff
@@ -24,11 +24,32 @@
 
 
 class FuelCodePermissions(permissions.BasePermission):
-    """Used by Viewset to check permissions for API requests"""
+    """
+    Used by Viewset to check permissions for API requests
+    This check is made before has_object_permission
+    If there's any permission check needed for POST, it has to be in here
+    """
 
     def has_permission(self, request, view):
-        return True
+        if request.user.has_perm('FUEL_CODES_MANAGE'):
+            return True
+
+        if request.method != 'POST' and \
+                request.user.has_perm('FUEL_CODES_VIEW'):
+            return True
+
+        return False
 
     def has_object_permission(self, request, view, obj):
-        """Check permissions When an object does exist (PUT, GET)"""
-        return True
+        """
```

Comment:
has_permission() is called by drf before has_object_permission() -- if has_permission() returns False, has_object_permission() will not run, so this is redundant